src/modules/Dashboard/components/TakeNowModal/TakeNowModal.tsx

```diff
@@ -0,0 +1,180 @@
+import { Box } from '@mui/material';
+
+import { Modal, SubmitBtnVariant } from 'shared/components';
+import { auth, workspaces } from 'redux/modules';
+import { StyledHeadline } from 'shared/styles';
+import { useAsync } from 'shared/hooks';
+import { DEFAULT_ROWS_PER_PAGE } from 'shared/consts';
+import { getWorkspaceRespondentsApi } from 'api';
+import { joinWihComma } from 'shared/utils';
+import { AutocompleteOption } from 'shared/components/FormComponents';
+
+import {
+  OpenTakeNowModalOptions,
+  TakeNowModalProps,
+  UseTakeNowModalProps,
+} from './TakeNowModal.types';
+import { StyledTakeNowModalContent, StyledTakeNowModalHeader } from './TakeNowModal.styles';
+import { StyledImageContainer, StyledImg } from '../ActivitySummaryCard/ActivitySummaryCard.styles';
+import { LabeledDropdown } from './LabeledDropdown/LabeledDropdown';
+import { ParticipantsData } from '../../features/Participants';
+import { BaseActivity } from '../ActivityGrid';
+
+export const useTakeNowModal = ({ dataTestId }: UseTakeNowModalProps) => {
+  const { t } = useTranslation('app');
+  const [activity, setActivity] = useState<BaseActivity | null>(null);
+  const [participants, setParticipants] = useState<AutocompleteOption[]>([]);
+  const [participantsAndTeamMembers, setParticipantsAndTeamMembers] = useState<
+    AutocompleteOption[]
+  >([]);
+  const [defaultSubject, setDefaultSubject] = useState<AutocompleteOption | null>(null);
+  const [defaultParticipant, setDefaultParticipant] = useState<AutocompleteOption | null>(null);
+  const { ownerId } = workspaces.useData() || {};
+  const userData = auth.useData();
+  const { appletId } = useParams();
+
+  const participantToOption = useCallback((participant: ParticipantsData['result'][0]) => {
+    const stringNicknames = joinWihComma(participant.nicknames, true);
+    const stringSecretIds = joinWihComma(participant.secretIds, true);
+    const id = participant.id ?? participant.details[0].subjectId;
+    let label = stringSecretIds;
+    if (stringNicknames) {
+      label = `${stringSecretIds} (${stringNicknames})`;
+    }
+
+    return { id, label };
+  }, []);
+
+  const { execute: getParticipants } = useAsync(getWorkspaceRespondentsApi, (response) => {
+    if (response?.data) {
+      const data = (response.data as ParticipantsData).result;
+      setParticipants(data.map(participantToOption));
+    }
+  });
+
+  useEffect(() => {
+    if (appletId) {
+      getParticipants({
+        params: {
+          ownerId,
+          appletId,
+          limit: DEFAULT_ROWS_PER_PAGE,
+        },
+      });
+    }
+
+    if (userData) {
+      // For now, the admin is the only member of this list
+      let whoIsResponding = `${userData.user.id}`;
+      if (userData.user.firstName) {
+        whoIsResponding += ` (${userData.user.firstName}`;
+        if (userData.user.lastName) {
+          whoIsResponding += ` ${userData.user.lastName[0].toUpperCase()}.`;
+        }
+        whoIsResponding += ')';
+      }
```

Comment:
We should probably confirm what the grey text is meant to be showing [in Figma](https://www.figma.com/file/6BNx63peNVzrHAF1FoZ78x/Multi-Informant-2024?type=design&node-id=3915-232753&mode=design&t=Ujj22OLyVsESGTcK-4). First name and abbreviated last name make intuitive sense, however the existing Respondents screen, as well as the new Participants screen, currently show nickname, not first & last name. There are many screens where a summarized version of a participant is shown, which I anticipated in my PR #1652 by creating a component, `ParticipantSnippet` to use for this very purpose (though maybe I should have called it `UserSnippet`). I may have assumed wrong what props it should take, but I settled with secret user ID, nickname, and tag, since that's what we're already showing in the Respondents and Participants tables. We should at least get some agreement as to the purpose of the `nickname` property and make sure to use it as intended in the UI.

Further to this: I noticed you opted to format these dropdowns in plain text, without the rich text formatting shown in Figma:

<img width="493" alt="Screenshot 2024-04-15 at 5 33 24 PM" src="https://github.com/ChildMindInstitute/mindlogger-admin/assets/1282772/95c7140a-21ac-4c47-8b65-3047cc9ad9ae">

I get that supporting this styling this could be a bit of a chore and you may have wanted to avoid it in your PR, but is there a separate ticket to take care of it?

Reply:
Back when I was implementing this part of the modal, the dropdown list looked like this

<img width="320px" src="https://github.com/ChildMindInstitute/mindlogger-admin/assets/14842108/36453356-8558-4672-9813-bb8f6d03de0a">

I hadn't updated my implementation because the design was still changing and I assumed I should work with the version that was there before. Although admittedly, I omitted the avatar icon for the same reason.

I think the question about nickname vs first and last name is a fair one. I'll ask about it in Slack, and I'll just update the component to match the current design

Reply:
Looks like [these are actually nicknames](https://www.figma.com/file/6BNx63peNVzrHAF1FoZ78x?type=design&node-id=3915-232728&mode=dev#770527901), so I'm going to steal your `ParticipantSnippet` component